### PR TITLE
added files for Collectd 5.6 support

### PR DESCRIPTION
**Pull request by tokyowizard**
CentOS 7 has updated the default version of collectd from 5.5 to 5.6.

This PR provides the `collectd.conf` and `librato.conf` for collectd v5.6.

The `collectd.conf` is taken from the default install of CentOS 7 with the same settings from the 5.5 file.

### CODE REVIEW

**Review by MiLk — CHANGES_REQUESTED** [review]
Please stop adding a file for each version.
Instead, use jinja to change what need to be changed inside the template file according to the version.

For the main config file, you can remove everything concerning plugins. What is not commented about plugins should be moved to a different file to be re used across versions.

And disable `AutoLoadPlugin`.

**Comment by tokyowizard** [comment]
There is now just 1 collectd.conf.j2 template file with jinja that include the changes according to the version.

All the plugins have been moved to their own files.

And `AutoLoadPlugin` is disabled.

File: tasks/main.yml [hunk 1]
```diff
@@ -38,7 +39,7 @@
 
 - name: "Upload Librato collectd config"
   template: >
-    src=librato.conf.{{ installed_version }}.j2
+    src=plugin-configs/librato.conf.j2
```
Comment (MiLk) [inline tasks/main.yml]: Can you not add librato to the plugin list to use only one block?

File: tasks/main.yml [hunk 2]
```diff
@@ -50,13 +51,40 @@
     - collectd
     - libs
 
+- name: "Upload coretemp plugin config"
+  template: >
+    src=plugin-configs/exec.conf.j2
+    dest={{ collectd_d_config_dir }}/exec.conf.j2
+    mode=0644
+  when: monitor_coretemp|bool
```
Comment (MiLk) [inline tasks/main.yml]: Use the conditional inside the file. It's ok to upload an empty file.

File: tasks/main.yml [hunk 3]
```diff
@@ -50,13 +51,40 @@
...
+    - collectd
+    - files
+    - configs
+
+- name: "Upload graphite plugin config"
+  template: >
+    src=plugin-configs/write_graphite.conf.j2
+    dest={{ collectd_d_config_dir }}/write_graphite.conf.j2
+    mode=0644
+  when: use_graphite|bool and collectd_forwarder is defined and collectd_forwarder == inventory_hostname
```
Comment (MiLk) [inline tasks/main.yml]: Same as coretemp.

File: templates/collectd.conf.j2 [hunk 4]
```diff
@@ -22,15 +22,15 @@ FQDNLookup   false
 # when an appropriate <Plugin ...> block is encountered.                     #
 # Disabled by default.                                                       #
 #----------------------------------------------------------------------------#
-AutoLoadPlugin true
-
+#AutoLoadPlugin false
```
Comment (MiLk) [inline templates/collectd.conf.j2]: Remove the comment. Set it explicitly to false.

File: templates/collectd.conf.j2 [hunk 5]
```diff
@@ -69,22 +68,22 @@ LoadPlugin syslog
...
+{% if installed_version|version_compare("5.5", ">=") %}#<Plugin log_logstash>
 #	LogLevel info
 #	File "/var/log/collectd.json.log"
 #</Plugin>
-
-<Plugin syslog>
-	LogLevel info
-</Plugin>
+{% endif %}
+#<Plugin syslog>
```
Comment (MiLk) [inline templates/collectd.conf.j2]: You probably need to leave the configuration for syslog plugin.
You should move it just after the `LoadPlugin syslog` line.

File: templates/collectd.conf.j2 [hunk 6]
```diff
@@ -69,22 +68,22 @@ LoadPlugin syslog
...
+#	LogLevel info
+#</Plugin>
 
 ##############################################################################
```
Comment (MiLk) [inline templates/collectd.conf.j2]: Remove everything from here to`Include "/etc/collectd.d"`.
It should be only comments.

File: templates/plugin-configs/plugin.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,9 @@
+LoadPlugin vmem
```
Comment (MiLk) [inline templates/plugin-configs/plugin.conf.j2]: That file doesn't seem to be used.

**Review by MiLk — CHANGES_REQUESTED** [review]

File: templates/collectd.conf.j2 (hunk 6 above)
Comment (tokyowizard) [reply templates/collectd.conf.j2]: I've removed all uncommented `LoadPlugin` and into its own .conf file, which were the defaults (`syslog`,`cpu`,`interface`,`load`,`memory`).
The rest are only comments.

File: README.md [hunk 8]
```diff
@@ -76,6 +76,26 @@ collectd1.somedc.prod         ansible_ssh_host=10.0.1.111   private_ip=172.16.1.
 
 ### Plugin configuration
 
+The `collectd_load_plugins` variable defines the plugins to be loaded. Default plugins are below:
```
Comment (MiLk) [inline README.md]: The name should be `collectd_plugins`.

File: README.md [hunk 9]
```diff
@@ -76,6 +76,26 @@ collectd1.somedc.prod         ansible_ssh_host=10.0.1.111   private_ip=172.16.1.
...
+
+```
+LoadPlugin syslog
```
Comment (MiLk) [inline README.md]: Why not use a list with just the plugin names?

File: templates/plugin-configs/python.conf.j2 [hunk 10]
```diff
@@ -0,0 +1,22 @@
+LoadPlugin python
```
Comment (MiLk) [inline templates/plugin-configs/python.conf.j2]: Why is it outside of the condition?

File: templates/plugin-configs/python.conf.j2 [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+{% if collectd_forwarder is defined and collectd_forwarder == inventory_hostname  %}
+{% if use_graphite|bool %}
+<LoadPlugin write_graphite>
```
Comment (MiLk) [inline templates/plugin-configs/python.conf.j2]: It seems to be a duplicate of `templates/plugin-configs/write_graphite.conf.j2`

File: templates/collectd.conf.j2 [hunk 12]
```diff
@@ -69,22 +71,19 @@ LoadPlugin syslog
...
 #	LogLevel info
 #	File "/var/log/collectd.json.log"
 #</Plugin>
-
-<Plugin syslog>
-	LogLevel info
-</Plugin>
+{% endif %}
 
 ##############################################################################
```
Comment (MiLk) [inline templates/collectd.conf.j2]: Please remove all the commented configuration from that point.
It's unwanted noise and make it harder to maintain.

**Review by MiLk — CHANGES_REQUESTED** [review]
Don't squash your PR, I feel like I have to review the 1895 lines you changed every time.

**Comment by tokyowizard** [comment]
Thank you for the comments. I've made the fixes. When you think it's good, shall I squash it before the PR is merged?

**Review by MiLk — APPROVED** [review]

**Pull request merged**